ff/serving/provider/dynamodb.go

```diff
@@ -0,0 +1,249 @@
+package provider
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	sn "github.com/mrz1836/go-sanitize"
+	"strconv"
+)
+
+type dynamodbTableKey struct {
+	Prefix, Feature, Variant string
+}
+
+func (t dynamodbTableKey) String() string {
+	marshalled, _ := json.Marshal(t)
+	return string(marshalled)
+}
+
+type dynamodbOnlineStore struct {
+	client *dynamodb.DynamoDB
+	prefix string
+	BaseProvider
+}
+
+type dynamodbOnlineTable struct {
+	client    *dynamodb.DynamoDB
+	key       dynamodbTableKey
+	valueType ValueType
+}
+
+type dynamodbItem struct {
+	Entity string `dynamodbav:"Entity"`
+	Value  string `dynamodbav:"FeatureValue"`
+}
+
+type Metadata struct {
+	Tablename string `dynamodbav:"Tablename"`
+	Valuetype string `dynamodbav:"ValueType"`
+}
+
+func dynamodbOnlineStoreFactory(serialized SerializedConfig) (Provider, error) {
+	dynamodbConfig := &DynamodbConfig{}
+	if err := dynamodbConfig.Deserialize(serialized); err != nil {
+		return nil, err
+	}
+	if dynamodbConfig.Prefix == "" {
+		dynamodbConfig.Prefix = "Featureform_table__"
+	}
+	return NewDyanamodbOnlineStore(dynamodbConfig), nil
+}
+
+func NewDyanamodbOnlineStore(options *DynamodbConfig) *dynamodbOnlineStore {
+	config := &aws.Config{
+		Endpoint: &options.Addr,
+		Region:   aws.String(options.Region),
+	}
+	sess := session.Must(session.NewSession(config))
+	dynamodbClient := dynamodb.New(sess)
+	CreateMetadataTable(dynamodbClient)
+	return &dynamodbOnlineStore{dynamodbClient, options.Prefix, BaseProvider{
+		ProviderType:   DynamoDBOnline,
+		ProviderConfig: options.Serialized(),
+	},
+	}
+}
+
+func (store *dynamodbOnlineStore) AsOnlineStore() (OnlineStore, error) {
+	return store, nil
+}
+
+func CreateMetadataTable(dynamodbClient *dynamodb.DynamoDB) {
```

Comment:
Whats this metadatatable for? Is it storing the names of the feature tables?

Reply:
The names of the feature tables and the value type associated with them. Primarily for the GetTable() function